### PR TITLE
tokenize_shuffle.py: tar files as input format

### PR DESCRIPTION
**Pull request by iejMac**

### CODE REVIEW

**Comment by iejMac** [comment]
replacing https://github.com/mlfoundations/open_lm/pull/66/files

File: open_lm/datapreprocess/ray/tokenize_shuffle.py [hunk 1]
```diff
@@ -313,6 +341,9 @@ def load_tokenizer(tokenizer):
     if tokenizer == "EleutherAI/gpt-neox-20b":
         enc = GPTNeoXTokenizerFast.from_pretrained("EleutherAI/gpt-neox-20b")
         return (lambda x: enc(x).input_ids, enc.vocab_size)
+    elif tokenizer == "none":
+        enc = lambda x: x
+        return (enc, -1)
```
Comment (iejMac) [inline open_lm/datapreprocess/ray/tokenize_shuffle.py:346]: not sure what to do for this. Maybe we can have the "tokenizer" arg double as a "vocab_size" arg for pre-tokenized data and then we just check `elif tokenizer.isnum()`. Not sure if this is clean

**Comment by iejMac** [comment]
this is pretty much good to go minus that one point about the "vocab_size" output of load_tokenizer

cc: @GeorgiosSmyrnis @Vaishaal

File: open_lm/datapreprocess/ray/tokenize_shuffle.py (hunk 1 above)
Comment (iejMac) [reply open_lm/datapreprocess/ray/tokenize_shuffle.py]: I tried something else, lmk how that it works:

If vocab_size is ever passed in that must mean tokenizer won't be needed so we don't load the tokenizer and instead return (lambda x: x, args.vocab_size)

**Comment by Vaishaal** [comment]
lgtm!

**Review by Vaishaal — APPROVED** [review]
LGTM

**Pull request merged**